src/Microsoft.ML.Data/Transforms/NormalizerStaticExtensions.cs

```diff
@@ -0,0 +1,366 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// Extension methods for static pipelines for normalization of data.
+    /// </summary>
+    public static class NormalizerStaticExtensions
+    {
+        private const long MaxTrain = Normalizer.Defaults.MaxTrainingExamples;
+        private const bool FZ = Normalizer.Defaults.FixZero;
+
+        /// <summary>
+        /// Learns an affine function based on the minimum and maximum, so that all values between the minimum and
+        /// maximum observed during fitting fall into the range of -1 to 1. Note that if values are later transformed
+        /// that are lower than the minimum, or higher than the maximum, observed during fitting, that the output
+        /// values may be outside the range of -1 to 1.
+        /// </summary>
+        /// <param name="input">The input column.</param>
+        /// <param name="fixZero">If set to <c>false</c>, then the observed minimum and maximum during fitting
+        /// will map to -1 and 1 respectively, exactly. If however set to <c>true</c>, then 0 will always map to 0.
+        /// This is valuable for the sake of sparsity preservation, if normalizing sparse vectors.</param>
+        /// <param name="maxTrainingExamples">When gathering statistics only look at most this many examples.</param>
+        /// <param name="onFit">A delegate that can be called whenever the function is fit, with the learned slopes
+        /// and, if <paramref name="fixZero"/> is <c>false</c>, the offsets as well.</param>
+        /// <remarks>Note that the statistics gathering and normalization is done independently per slot of the
+        /// vector values.</remarks>
+        /// <returns>The normalized column.</returns>
+        public static NormVector<float> NormalizeByMinMax(
```

Comment:
The promulgation of extension methods is I think angering @Zruty0 . In particular the problem may be that if someone types `Normalize` on a `Vector<float>` or `Vector<double>` they will have at the present four distinct normalization strategies presented to them. Historically we have, for no particularly reason, considered `MinMax` to be the "default" normalizer, somehow. But in the API the idea of a default is less important. `NormalizeByBin` is the first alphabetically, which is by itself rather unfortunate since it is by far the one that takes the most resources (making, as it does, practically a copy of the dataset -- so it is in some way the one we want users to be most deliberate about using).

A few things are possible:

1. Rename `NormalizeMinMax` to `Normalize`, thereby emphasizing it is essentially the default normalizer. We will be clear about this in the documentation.

2. Similar to the above, but keeping both, with the idea that one is an alias to the other.

3. Hide all normalizers except `NormalizeMinMax` behind another namespace, e.g., keep `NormalizeMinMax` in say `Microsoft.ML.Transforms` but store the other normalizers in `Microsoft.ML.Transforms. Normalizers`. 

I am personally most in favor of 1. 2 and 3 both seems odd the more I think on it.

Reply:
I am OK with 1

---
In reply to: [215715197](https://github.com/dotnet/machinelearning/pull/843#discussion_r215715197) [](ancestors = 215715197)

Reply:
Renamed to `Normalize`.